### PR TITLE
Find Projects using jdbi

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java [hunk 1]
```diff
@@ -343,4 +379,358 @@ public FutureProjectDAO(
     return permissionCheck.thenCompose(
         unused -> artifactHandler.getUrlForArtifact(request), executor);
   }
+
+  public InternalFuture<FindProjects.Response> findProjects(FindProjects request) {
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java:390]: This method has hundreds of lines. It's even hard to read with the indentation! Let's split it up.

File: backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java [hunk 2]
```diff
@@ -343,4 +379,358 @@ public FutureProjectDAO(
...
+          .clearAttributes()
+          .addAllAttributes(
+              project.getAttributesList().stream()
+                  .sorted(Comparator.comparing(KeyValue::getKey))
+                  .collect(Collectors.toList()))
+          .clearArtifacts()
+          .addAllArtifacts(
+              project.getArtifactsList().stream()
+                  .sorted(Comparator.comparing(Artifact::getKey))
+                  .collect(Collectors.toList()));
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java:672]: Could we just place this responsibility into `TagsHandler` and associated? Each one returns a sorted list. That way, we ensure they will always return sorted no matter which object is being queried.

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java:672]: Move this logic to individual handlers in PR: https://github.com/VertaAI/modeldb/pull/2792
Handler logic was added in PR https://github.com/VertaAI/modeldb/pull/2792 so not modified here.

File: backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java:390]: Splatted findProjects function in multiple functions

**Review by conradoverta — APPROVED** [review]

**Pull request merged**